src/test/java/org/junit/runners/CustomBlockJUnit4ClassRunnerTest.java

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+
+package org.junit.runners;
+
+import static org.junit.Assert.*;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.Description;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.Statement;
+
+/**
+ * Tests that verify proper behavior for custom runners that extend
+ * {@link BlockJUnit4ClassRunner}.
+ *
+ * @author Sam Brannen
+ * @since 4.13
+ */
+public class CustomBlockJUnit4ClassRunnerTest {
+
+	/**
+	 * This code has been adapted from the Spring Framework.
+	 */
+	@Test
+	public void exceptionsFromMethodBlockMustNotResultInUnrootedTests() throws Exception {
+		Class<CustomBlockJUnit4ClassRunnerTestCase> testClass = CustomBlockJUnit4ClassRunnerTestCase.class;
+		TrackingRunListener listener = new TrackingRunListener();
+		RunNotifier notifier = new RunNotifier();
+		notifier.addListener(listener);
+
+		new CustomBlockJUnit4ClassRunner(testClass).run(notifier);
+		assertEquals("Number of tests started for [" + testClass + "].", 2, listener.getTestStartedCount());
+		assertEquals("Number of tests ignored for [" + testClass + "].", 0, listener.getTestIgnoredCount());
+		assertEquals("Number of assumption failures for [" + testClass + "].", 0, listener.getTestAssumptionFailureCount());
+		assertEquals("Number of test failures for [" + testClass + "].", 1, listener.getTestFailureCount());
+		assertEquals("Number of tests finished for [" + testClass + "].", 2, listener.getTestFinishedCount());
+	}
+
+
+	@Ignore("This test case is run manually by the enclosing test class")
+	public static final class CustomBlockJUnit4ClassRunnerTestCase {
+		@Test public void shouldPass() { /* no-op */ }
+		@Test public void throwException() { /* no-op */ }
+	}
+
+	/**
+	 * Custom extension of {@link BlockJUnit4ClassRunner} that always throws
+	 * an exception from the {@code methodBlock()} if a test method is named
+	 * exactly {@code "throwException"}.
+	 */
+	static class CustomBlockJUnit4ClassRunner extends BlockJUnit4ClassRunner {
+
+		public CustomBlockJUnit4ClassRunner(Class<?> testClass) throws InitializationError {
+			super(testClass);
+		}
+
+		@Override
+		protected Statement withBefores(FrameworkMethod method, Object target, Statement statement) {
+			if ("throwException".equals(method.getName())) {
+				throw new RuntimeException("throwException() test method invoked");
+			}
+			return super.withBefores(method, target, statement);
+		}
+	}
+
+	/**
+	 * Simple {@link RunListener} which tracks how many times certain JUnit callback
+	 * methods were called: only intended for the integration test suite.
+	 * <p>This code has been copied directly from the Spring Framework.
+	 * @author Sam Brannen
```

Comment:
I am in fact the author of the code in question, but as far as I understand that is irrelevant since the ASL license header is included in the file along with the `@author` tags.

Now, I am _not_ a lawyer, but I don't think there is any issue here since:

1. The original ASL 2.0 license header is present in the file I submitted.
2. The code in question is not distributed per se with JUnit JARs; it resides in the test suite.